src/expr/proof_node_manager.cpp

```diff
@@ -154,7 +135,7 @@ std::shared_ptr<ProofNode> ProofNodeManager::mkScope(
   }
   else if (ac.size() < assumps.size())
   {
-    // always must remove duplicates
+    // remove duplicates to avoid redundant literals in clauses
```

Comment:
I imagine this is an optimization then? Was it not done before?

Reply:
It is an optimization that has always been done, it used to be required, now the code is robust to  if it weren't done (I think).

Reply:
The SAT solver silently eliminates such duplicates. What is puzzling me is just that this is being done here rather than in the user of this class.

Reply:
ah, so I think originally the ProofEqEngine was doing this itself. This code is actually based off a number of theories explain methods which were doing it themselves. I think it's a good idea to always do this here.

Reply:
If we are doing proofs, indeed. But if we are not? Since this is about the explanation assumptions itself I'd expect this to be made independent of proofs.

Reply:
The reason this is done here when proofs are enabled is because it is convenient to do so here, since we've computed `ac`.  Thus, we shouldn't require the user to clean this up themselves, since then we'd actually be converting to/from vector/unordered_set twice.

If proof are disabled, then it is up to the user eliminate duplicates themselves.  That is a separate issue.

Regardless, it's for performance reasons that we don't insist that duplicates are eliminated before calling mkScope.

Reply:
OK. In that case then I'd like to have this explicitly controlled via a parameter, like `doMinimize`, which would be `true` by default. What do you think?

Reply:
Is there a use case for this?  If so, I can add this. However, I'd prefer not adding this.

Reply:
Testing efficiency, which I assume is the same use case of `doMinimize`?